raft/raft.go

```diff
@@ -750,6 +790,25 @@ func stepLeader(r *raft, m pb.Message) {
 		if pr.Match < r.raftLog.lastIndex() {
 			r.sendAppend(m.From)
 		}
+
+		if r.readOnly.option != ReadOnlySafe || len(m.Context) == 0 {
+			return
+		}
+
+		ackCount := r.readOnly.recvAck(m)
```

Comment:
originally, no. we never send heartbeat with same ctx twice. i made a change right now, which might introduce dup. i have added the debup.